src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1468 @@
+// buffer size is what we tell the pipe it is, so we have room
+// for out of band data etc.
+const MAX_FRAGMENT_SIZE: usize = 64 * 1024;
+
+// Size of the pipe's write buffer, with excess room for the header.
+const PIPE_BUFFER_SIZE: usize = MAX_FRAGMENT_SIZE + 4 * 1024;
+
+#[allow(non_snake_case)]
+fn GetLastError() -> u32 {
+    unsafe {
+        kernel32::GetLastError()
+    }
+}
+
+pub fn channel() -> Result<(OsIpcSender, OsIpcReceiver),WinError> {
+    let pipe_id = make_pipe_id();
+    let pipe_name = make_pipe_name(&pipe_id);
+
+    let receiver = try!(OsIpcReceiver::new_named(&pipe_name));
+    let sender = try!(OsIpcSender::connect_named(&pipe_name));
+
+    Ok((sender, receiver))
+}
+
+// Holds data len and out-of-band data len
+struct MessageHeader(u32, u32);
+
+impl MessageHeader {
+    fn size() -> usize {
+        mem::size_of::<MessageHeader>()
+    }
+
+    fn total_message_bytes_needed(&self) -> usize {
+        MessageHeader::size() + self.0 as usize + self.1 as usize
+    }
+}
+
+struct Message<'data> {
+    data_len: usize,
+    oob_len: usize,
+    bytes: &'data [u8],
+}
+
+impl<'data> Message<'data> {
+    fn from_bytes(bytes: &'data [u8]) -> Option<Message> {
+        if bytes.len() < MessageHeader::size() {
+            return None;
+        }
+
+        unsafe {
+            let ref header = *(bytes.as_ptr() as *const MessageHeader);
+            if bytes.len() < header.total_message_bytes_needed() {
+                return None;
+            }
+
+            Some(Message {
+                data_len: header.0 as usize,
+                oob_len: header.1 as usize,
+                bytes: &bytes[0..header.total_message_bytes_needed()],
+            })
+        }
+    }
+
+    fn data(&self) -> &[u8] {
+        &self.bytes[MessageHeader::size()..(MessageHeader::size() + self.data_len)]
+    }
+
+    fn oob_bytes(&self) -> &[u8] {
+        &self.bytes[(MessageHeader::size() + self.data_len)..]
+    }
+
+    fn oob_data(&self) -> Option<OutOfBandMessage> {
+        if self.oob_len > 0 {
+            let oob = bincode::serde::deserialize::<OutOfBandMessage>(self.oob_bytes())
+                .expect("Failed to deserialize OOB data");
+            // On windows, since we're duplicating handles intended for a specific
+            // process.
+            if oob.target_process_id != *CURRENT_PROCESS_ID {
+                panic!("Windows IPC channel received handles intended for process {}, but this is {}",
+                       oob.target_process_id, *CURRENT_PROCESS_ID);
+            }
+            Some(oob)
+        } else {
+            None
+        }
+    }
+
+    fn size(&self) -> usize {
+        MessageHeader::size() + self.data_len + self.oob_len
+    }
+}
+
+// If we have any channel handles or shmem segments,
+// then we'll send an OutOfBandMessage after
+// the data message.
+#[derive(Debug)]
+struct OutOfBandMessage {
+    target_process_id: u32, // the process ID of the intended target process
```

Comment:
Added explanation comment, and a TODO item with a couple possible mitigations/solutions.